src/Pate/Proof/Instances.hs

```diff
@@ -448,9 +449,22 @@ ppMemCellMap ::
   MapF.MapF (PMC.MemCell grnd arch) (PF.BlockSliceMemOp grnd) ->
   PP.Doc a
 ppMemCellMap dom cells = let
-  ppCells = mapMaybe (\(Pair cell v) -> ppCellVal dom cell v) $ MapF.toList cells
+  eqEntries (Pair cell1 _) (Pair cell2 _) = eqGroundCells cell1 cell2
+  ppCells = mapMaybe (\(Pair cell v) -> ppCellVal dom cell v) $ nubBy eqEntries $ MapF.toList cells
```

Comment:
How many cells do you expect here in general? How about the worst case? I'm asking because `nubBy` is infamously quadratic

Reply:
It's the cells that show up in the counter-example for a single slice, so maybe a dozen or so at most?